src/all/webtoons/src/eu/kanade/tachiyomi/extension/all/webtoons/WebtoonsSrc.kt

```diff
@@ -45,12 +46,44 @@ open class WebtoonsSrc(
             }
         }
         screen.addPreference(authorsNotesPref)
+
+        val maxQualityPref = SwitchPreferenceCompat(screen.context).apply {
+            key = USE_MAX_QUALITY_KEY
+            title = "Use maximum quality images"
+            summary = "Enable to load images in maximum quality."
+            setDefaultValue(false)
+
+            setOnPreferenceChangeListener { _, newValue ->
+                val checkValue = newValue as Boolean
+                preferences.edit().putBoolean(USE_MAX_QUALITY_KEY, checkValue).commit()
+            }
+        }
+        screen.addPreference(maxQualityPref)
     }
 
     private fun showAuthorsNotesPref() = preferences.getBoolean(SHOW_AUTHORS_NOTES_KEY, false)
+    private fun useMaxQualityPref() = preferences.getBoolean(USE_MAX_QUALITY_KEY, false)
 
     override fun pageListParse(document: Document): List<Page> {
-        var pages = document.select("div#_imageList > img").mapIndexed { i, element -> Page(i, "", element.attr("data-url")) }
+        var pages = document.select("div#_imageList > img").mapIndexed { i, element ->
+            val imageUrl = element.attr("data-url")
+
+            if (useMaxQualityPref()) {
+                // Return original if parsing fails
+                val httpUrl = imageUrl.toHttpUrlOrNull() ?: return@mapIndexed Page(i, "", imageUrl)
```

Comment:
parsing failing means url isn't valid http url, and that would not load afterwards too, so just use `toHttpUrl()`